File: python-app/seed.py
"""Creates the demo studio used by the Next.js version's seed script, for parity."""

from app import create_app
from auth import hash_password
from db import get_db
from utils import new_id

EMAIL = "demo@aperture-co.com"
PASSWORD = "password123"


def main():
    app = create_app()
    with app.app_context():
        database = get_db()
        existing = database.execute("SELECT 1 FROM users WHERE email = ?", (EMAIL,)).fetchone()
        if existing:
            print("Demo studio already seeded.")
            return

        user_id = new_id()
        studio_id = new_id()
        database.execute(
            "INSERT INTO users (id, email, password_hash, name) VALUES (?, ?, ?, ?)",
            (user_id, EMAIL, hash_password(PASSWORD), "Demo Photographer"),
        )
        database.execute(
            """INSERT INTO studios
                (id, owner_id, slug, brand_name, tagline, description, email, phone, location,
                 instagram, plan)
               VALUES (?, ?, 'aperture-co', 'Aperture & Co.', ?, ?, 'hello@aperture-co.com',
                       '+1 (555) 012-3456', 'Based in New York, available worldwide',
                       'https://instagram.com', 'pro')""",
            (
                studio_id,
                user_id,
                "Photography that feels like a memory you already had.",
                "Aperture & Co. is a photography studio specializing in portrait, graduation, and "
                "professional sessions — crafted with care from the first frame to the final print.",
            ),
        )
        for url, title, category in [
            ("/static/photos/graduation-seated.jpg", "Class of 2026", "Graduation"),
            ("/static/photos/graduation-headshot.jpg", "Studio Headshot", "Graduation"),
            ("/static/photos/professional-standing.jpg", "Professional Portrait", "Portrait"),
        ]:
            database.execute(
                "INSERT INTO photos (id, studio_id, url, title, category) VALUES (?, ?, ?, ?, ?)",
                (new_id(), studio_id, url, title, category),
            )
        database.commit()
        print(f"Seeded demo studio for user {EMAIL} (password: {PASSWORD})")


if __name__ == "__main__":
    main()

File: prisma/seed.ts
import bcrypt from "bcryptjs";
import { PrismaClient } from "@prisma/client";
import { PrismaBetterSqlite3 } from "@prisma/adapter-better-sqlite3";

const adapter = new PrismaBetterSqlite3({ url: "file:./prisma/dev.db" });
const db = new PrismaClient({ adapter });

async function main() {
  const email = "demo@aperture-co.com";
  const password = "password123";

  const existing = await db.user.findUnique({ where: { email } });
  if (existing) {
    console.log("Demo studio already seeded.");
    return;
  }

  const passwordHash = await bcrypt.hash(password, 10);

  const user = await db.user.create({
    data: {
      name: "Demo Photographer",
      email,
      passwordHash,
      studio: {
        create: {
          slug: "aperture-co",
          brandName: "Aperture & Co.",
          tagline: "Photography that feels like a memory you already had.",
          description:
            "Aperture & Co. is a photography studio specializing in portrait, graduation, and professional sessions — crafted with care from the first frame to the final print.",
          email: "hello@aperture-co.com",
          phone: "+1 (555) 012-3456",
          location: "Based in New York, available worldwide",
          instagram: "https://instagram.com",
          plan: "pro",
          photos: {
            create: [
              { url: "/photos/graduation-seated.jpg", title: "Class of 2026", category: "Graduation" },
              { url: "/photos/graduation-headshot.jpg", title: "Studio Headshot", category: "Graduation" },
              { url: "/photos/professional-standing.jpg", title: "Professional Portrait", category: "Portrait" },
            ],
          },
        },
      },
    },
  });

  console.log(`Seeded demo studio for user ${user.email} (password: ${password})`);
}

main()
  .catch((err) => {
    console.error(err);
    process.exit(1);
  })
  .finally(async () => {
    await db.$disconnect();
  });
